Fix for fits_image_viewer visualization plugin

francoismg opened the pull request:
The fits_image_viewer (used for visualization of FITS file) has been unusable on usegalaxy.eu and on some local instances.

Two differents bugs are causing this issue (more details can be found in this issue https://github.com/usegalaxy-eu/galaxy/issues/194) :

- The first one is only related to the usegalaxy.eu server that doesn't serve the correct mime type for web assembly file
- The second one is caused by the aladin lite js files not being copied to the correct location.

This fix makes the plugin relies on the university of strasbourg CDN (which should be a workaround for the server mime type issue) with a fallback to local aladin lite js files if the cdn were not to be available.

The second fix is to force the creation of the aladin lite folder directly on the build script in order to resolve the issue that on some instances the folder was not created correctly which caused some missing js files error 

@bgruening since the plugin is not usable right now on galaxy.eu we were wondering what kind of timeframe we should expect for the fix to be pushed to galaxy.eu prod, thanks

## How to test the changes?
- [X] Instructions for manual testing are as follows:
  1. Upload a fits file to galaxy (one should be present on the test-data folder)
  2. Use the visualization plugin to open the fits file (after a few second a map of space should be displayed with the content of the file in the middle) 

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

francoismg committed:
fix for folder location

francoismg committed:
added cdn + fallback

francoismg committed:
added cdn and fallback urls + refactor

dsavchenko committed:
fix no such directory in fits viewer build

dannon commented on config/plugins/visualizations/fits_image_viewer/package.json:
```diff
@@ -11,6 +11,6 @@
     },
     "scripts": {
-        "build": "cp -r node_modules/aladin-lite-galaxy static/dist && parcel build src/script.js -d static"
+        "build": "mkdir -p static/dist; cp -r node_modules/aladin-lite-galaxy static/dist/. && parcel build src/script.js -d static"
```
A more robust approach other plugins take is to use https://github.com/sindresorhus/cpy-cli -- that's what I'd recommend here.  Nora is a bit different in that it just stages and doesn't build, but it's still a good example: https://github.com/galaxyproject/galaxy/blob/dev/config/plugins/visualizations/nora/package.json#L16

francoismg replied:
thanks, I didn't find other plugins using it except nora like you mentionned but it seems good, it looks like it would prevent the difference in behavior between different instances we have been experiencing, will make the switch to cpy and update the pr

francoismg committed:
added cpy-cli dependency and updated build script

Following advice from galaxy maintainer :

- Switch to cpy-cli to use instead of cp (should prevent the differences in cp behaviour)
